### PR TITLE
 install EasyBuild as a module in base image + clean up GCC 9.3.0 image 

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: containers/Dockerfile.base-centos7 [hunk 1]
```diff
@@ -19,5 +21,15 @@ SHELL ["/bin/bash", "-l", "-c"]
...
-RUN ml --version && which -a eb && eb --version & eb --show-system-info
+RUN ml use /easybuild/modules/all && ml EasyBuild && ml --version && which -a eb && eb --version & eb --show-system-info
+# remove globally installed EasyBuild, we can use the module going forward
+USER root
+RUN pip3 uninstall -y easybuild
+# switch to 'easybuild' user
+USER easybuild
+WORKDIR /home/easybuild
+# disable output buffering in Python, so we see EasyBuild output as it progresses
+ENV PYTHONUNBUFFERED=TRUE
```
Comment (kniec) [inline containers/Dockerfile.base-centos7:35]: ENV can be set at the top and will be present no matter the `USER` currently used.
```
FROM busybox

ENV TEST=1
USER nobody
RUN echo $TEST
```

File: containers/Dockerfile.base-centos7 [hunk 2]
```diff
@@ -4,11 +4,13 @@ ARG EB_VER=4.2.1
 LABEL easybuild_version=${EB_VER}
 # install required packages (EPEL is required for Lmod)
 RUN yum install -y epel-release \
- && yum install -y bzip2 file gcc-c++ git Lmod make openssl-devel patch python3-pip unzip vim which
+ && yum install -y bzip2 file gcc-c++ git Lmod make openssl openssl-devel patch python3-pip unzip vim which rdma-core-devel
```
[the remark this answers is not in the document]
Comment (boegel) [reply containers/Dockerfile.base-centos7:7]: Having `rdma-core-devel` installed affects how OpenMPI is configured (`--with-verbs` or `--without-verbs`).

I'm happy to remove it, it's probably overkill here, but I don't think it hurts to have it installed?

[the remark this answers is not in the document]
Comment (kniec) [reply containers/Dockerfile.base-centos7:12]: just wonder if we need to install this in the base - but ok... premature optimization I'd say

File: containers/Dockerfile.base-centos7 [hunk 3]
```diff
@@ -19,5 +29,20 @@ SHELL ["/bin/bash", "-l", "-c"]
...
+RUN eb --install-latest-eb-release --prefix /scratch --installpath /easybuild \
+  && rm -rf /scratch/*
 # print info on Lmod & EasyBuild
-RUN ml --version && which -a eb && eb --version & eb --show-system-info
+RUN ml use /easybuild/modules/all \
+  && ml EasyBuild \
+  && ml --version \
+  && which -a eb \
+  && eb --version \
+  && eb --show-system-info
```
Comment (kniec) [inline containers/Dockerfile.base-centos7:41]: beautiful

Comment (boegel) [reply containers/Dockerfile.base-centos7:41]: 😍

File: containers/Dockerfile.base-centos7 (hunk 2 above)
[the remark this answers is not in the document]
Comment (boegel) [reply containers/Dockerfile.base-centos7:12]: Again, I don't think it hurts much, except for size of the container a little bit?

Stripping it out is not a huge deal for the tutorial, and if it's there you can add `rdma-core-devel` in another image on top of this one, and rebuild `OpenMPI` (and everything that depends on it...), if you really want IB support, but that's a hassle.

[the remark this answers is not in the document]
Comment (kniec) [reply containers/Dockerfile.base-centos7:12]: let's keep it for now... and put a pin in it for a later discussion

[the remark this answers is not in the document]
Comment (boegel) [reply containers/Dockerfile.base-centos7:12]: Open an issue on it in this repo

[the remark this answers is not in the document]
Comment (kniec) [reply containers/Dockerfile.base-centos7:12]: #10

File: containers/Dockerfile.base-centos7 [hunk 4]
```diff
@@ -19,5 +29,20 @@ SHELL ["/bin/bash", "-l", "-c"]
...
+# remove globally installed EasyBuild, we can use the module going forward
+USER root
+RUN pip3 uninstall -y easybuild
```
Comment (kniec) [inline containers/Dockerfile.base-centos7:44]: potential optimization #11

**Pull request merged**